### PR TITLE
Make sam ONNX exportable

### PR DESCRIPTION
**Pull request by fxmarty**
As per title, would be great to have this PR on the next release so that we can support the ONNX export (see https://github.com/huggingface/optimum/pull/995). This piece is the only blocking one.

### CODE REVIEW

File: src/transformers/models/sam/modeling_sam.py [hunk 1]
```diff
@@ -623,10 +623,15 @@ def _embed_points(self, points: torch.Tensor, labels: torch.Tensor, pad: bool) -
         input_shape = (self.input_image_size, self.input_image_size)
         point_embedding = self.shared_embedding(points, input_shape)
 
-        point_embedding[labels == -1] = 0.0
-        point_embedding[labels == -1] += self.not_a_point_embed.weight
```
Comment (fxmarty) [inline src/transformers/models/sam/modeling_sam.py:627]: By the way what's the point of these two steps rather than `point_embedding[labels == -1] = self.not_a_point_embed.weight`?

Comment (ArthurZucker) [reply src/transformers/models/sam/modeling_sam.py:627]: not sure either! Thanks for the catch.

**Review by ArthurZucker — COMMENTED** [review]
Looks good to me. Should you also add an ONXX config and enable the CI tests ? (not sure maybe tracing will work)

**Comment by fxmarty** [comment]
@ArthurZucker The PR is here: https://github.com/huggingface/optimum/issues/991

File: src/transformers/models/sam/modeling_sam.py (hunk 1 above)
Comment (fxmarty) [reply src/transformers/models/sam/modeling_sam.py:627]: Ok I'll simplify then

File: src/transformers/models/sam/modeling_sam.py [hunk 2]
```diff
@@ -623,10 +623,16 @@ def _embed_points(self, points: torch.Tensor, labels: torch.Tensor, pad: bool) -
...
-
-        point_embedding[labels == -10] = 0.0  # ignore points
+        # torch.whereh and expanding the labels tensor is required by the ONNX export
```
Comment (sgugger) [inline src/transformers/models/sam/modeling_sam.py:626]:
```suggestion
        # torch.where and expanding the labels tensor is required by the ONNX export
```

**Review by sgugger — APPROVED** [review]
Thanks for adding this!
